sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClientTest.java

```diff
@@ -125,6 +124,108 @@ void teardown() {
         messagesCaptor = null;
     }
 
+    /**
+     * Verifies that the correct Service Bus properties are set.
+     */
+    @Test
+    void verifyProperties() {
+        Assertions.assertEquals(ENTITY_NAME, sender.getEntityName());
+        Assertions.assertEquals(NAMESPACE, sender.getFullyQualifiedNamespace());
+    }
+
+    /**
+     * Verifies that an exception is thrown when we create a batch with null options.
+     */
+    @Test
+    void createBatchNull() {
+        Assertions.assertThrows(NullPointerException.class, () -> sender.createBatch(null));
+    }
+
+    /**
+     * Verifies that the default batch is the same size as the message link.
+     */
+    @Test
+    void createBatchDefault() {
+        // Arrange
+        when(connection.createSendLink(eq(ENTITY_NAME), eq(ENTITY_NAME), any(AmqpRetryOptions.class)))
+            .thenReturn(Mono.just(sendLink));
+        when(sendLink.getLinkSize()).thenReturn(Mono.just(MAX_MESSAGE_LENGTH_BYTES));
+
+        // Act & Assert
+        StepVerifier.create(sender.createBatch())
+            .assertNext(batch -> {
+                Assertions.assertEquals(MAX_MESSAGE_LENGTH_BYTES, batch.getMaxSizeInBytes());
+                Assertions.assertEquals(0, batch.getCount());
+            })
+            .verifyComplete();
+    }
+
+    /**
+     * Verifies we cannot create a batch if the options size is larger than the link.
+     */
+    @Test
+    void createBatchWhenSizeTooBig() {
+        // Arrange
+        int maxLinkSize = 1024;
+        int batchSize = maxLinkSize + 10;
+
+        final AmqpSendLink link = mock(AmqpSendLink.class);
+        when(link.getLinkSize()).thenReturn(Mono.just(maxLinkSize));
+
+        when(connection.createSendLink(eq(ENTITY_NAME), eq(ENTITY_NAME), eq(retryOptions)))
+            .thenReturn(Mono.just(link));
+
+        // This event is 1024 bytes when serialized.
+        final CreateBatchOptions options = new CreateBatchOptions().setMaximumSizeInBytes(batchSize);
+
+        // Act & Assert
+        StepVerifier.create(sender.createBatch(options))
+            .expectError(IllegalArgumentException.class)
+            .verify();
+    }
+
+    /**
+     * Verifies that the producer can create a batch with a given {@link CreateBatchOptions#getMaximumSizeInBytes()}.
+     */
+    @Test
+    void createsEventDataBatchWithSize() {
```

Comment:
Should we use the term event or  `message` for Service bus? 
`createsMessageBatchWithSize`